tests/nativeapp/test_sf_sql_facade.py

```diff
@@ -0,0 +1,841 @@
+    mock_parent.attach_mock(mock_execute_queries, "mock_execute_queries")
+
+    all_execute_calls = [
+        mock.call.mock_execute_query("select current_role()"),
+        mock.call.mock_execute_query("use role mock_role"),
+        mock.call.mock_execute_query("select current_warehouse()"),
+        mock.call.mock_execute_query("use warehouse mock_wh"),
+        mock.call.mock_execute_query("select current_database()"),
+        mock.call.mock_execute_query("use database mock_db"),
+        mock.call.mock_execute_queries(mock_script),
+        mock.call.mock_execute_query("use database old_db"),
+        mock.call.mock_execute_query("use warehouse old_wh"),
+        mock.call.mock_execute_query("use role old_role"),
+    ]
+
+    # Act
+    sql_facade.execute_user_script(mock_script, mock_script_name, role, wh, database)
+
+    # Assert
+    assert mock_execute_query.mock_calls == expected
+    assert mock_execute_queries.mock_calls == [mock.call(mock_script)]
+    # Assert - order of calls
+    mock_parent.assert_has_calls(all_execute_calls)
+
+
+@mock.patch(SQL_EXECUTOR_EXECUTE)
+@mock.patch(SQL_EXECUTOR_EXECUTE_QUERIES)
+def test_execute_no_db(mock_execute_queries, mock_execute_query, mock_cursor):
+    # Arrange
+    mock_script = "-- my comment\nselect 1;\nselect 2;"
+    mock_script_name = "test-user-sql-script.sql"
+    role = "mock_role"
+    wh = "mock_wh"
+    side_effects, expected = mock_execute_helper(
+        [
+            (
+                mock_cursor([("old_role",)], []),
+                mock.call("select current_role()"),
+            ),
+            (None, mock.call("use role mock_role")),
+            (
+                mock_cursor([("old_wh",)], []),
+                mock.call("select current_warehouse()"),
+            ),
+            (None, mock.call("use warehouse mock_wh")),
+            (None, mock.call("use warehouse old_wh")),
+            (None, mock.call("use role old_role")),
+        ]
+    )
+    mock_execute_query.side_effect = side_effects
+
+    mock_parent = mock.Mock()
+    mock_parent.attach_mock(mock_execute_query, "mock_execute_query")
+    mock_parent.attach_mock(mock_execute_queries, "mock_execute_queries")
+
+    all_execute_calls = [
+        mock.call.mock_execute_query("select current_role()"),
+        mock.call.mock_execute_query("use role mock_role"),
+        mock.call.mock_execute_query("select current_warehouse()"),
+        mock.call.mock_execute_query("use warehouse mock_wh"),
+        mock.call.mock_execute_queries(mock_script),
+        mock.call.mock_execute_query("use warehouse old_wh"),
+        mock.call.mock_execute_query("use role old_role"),
+    ]
+
+    # Act
+    sql_facade.execute_user_script(mock_script, mock_script_name, role, wh)
+
+    # Assert
+    assert mock_execute_query.mock_calls == expected
+    assert mock_execute_queries.mock_calls == [mock.call(mock_script)]
+    # Assert - order of calls
+    mock_parent.assert_has_calls(all_execute_calls)
```

Comment:
these tests are mostly boilerplate. Would love to simplify, especially since we're checking the same thing two different ways.

Reply:
I'm with you. Didn't like writing these tests either. I'll think on it and make a better solution/helper.